### ISSUE TITLE
Regression: plotting legend hangs and cannot read from text file

### ISSUE THREAD

**Issue by adigitoleo** [GenericMappingTools/GMT.jl#826]
I had some code that worked before ca. version 0.38?

```
    legend(
        legendfile,
        position = (; anchor = :BL),
        box = (; pen = :thinner, fill = :white),
    )
```

This now complains that option `-D` is not set, which is aliased to `position`. I tried adding `inside = :BL` to the position NamedTuple, but then the code hangs indefinitely.

I'll try to come up with a proper minimal reproducible example.

**Comment by adigitoleo** [comment]
Reproduced on v0.40.1 with:

```
julia> using GMT

julia> gmtbegin("test_legend.png")

julia> coast(region = (110, 210, -35, 45), projection = :Mercator, frame = :auto, land = :dimgray, shore = :thinner)

julia> arrows(hcat(rand(110:210, 10), rand(-35:45, 10), zeros(10), rand(-90:90, 10), rand(1:3, 10)), linewidth = :thickest, linecolor = :black, vecmap = (; justify = :center), transparency = 33)

julia> open("legend.txt", append = true) do file
           println(file, "S 0.25c - 0.33 black 2p 0.75c 1s delay time")
       end

julia> legend("legend.txt", position = (; inside = :BL, anchor = :BL), box = (; pen = :thinner, fill = :white))
^Z
zsh: suspended  julia --banner=no --project=$PWD
admin@hades ~/Documents/PhD/code/Maps
%zsh> cat legend.txt                                                                              [bg:1]
S 0.25c - 0.33 black 2p 0.75c 1s delay time
admin@hades ~/Documents/PhD/code/Maps
%zsh> fg                                                                                          [bg:1]
[1]  + continued  julia --banner=no --project=$PWD


julia> gmtend(show = true)
```

Produced a box with no contents, and legend call had to be interrupted with Ctrl-D.

![image](https://user-images.githubusercontent.com/34595875/152707003-26e1632f-b97c-4a73-9a1b-dd663e286860.png)

**Comment by joa-quim** [comment]
... did a lot of improvements in ``legend`` some versions ago.
Seems that not all were improvements.

**Comment by adigitoleo** [comment]
Possibly caused by lack of `width` in `position` NamedTuple. This even simpler snippet causes a segfault:

```
julia> legend(
           region = (0, 10, 0, 8),
           (;
               symbol1 = (;
                   dx_left = 0.25,
                   marker = "-",
                   size = 0.33,
                   pen = 2,
                   fill = :black,
                   dx_right = 0.75,
                   text = "1s delay time",
               ),
               symbol2 = (;
                   dx_left = 0.25,
                   marker = :star,
                   size = 0.5,
                   pen = :false,
                   fill = :yellow,
                   dx_right = 0.75,
                   text = "Epicentre",
               ),
           ),
           position = (; inside = :BL, anchor = :BL),
           box = (; pen = 0.5),
       )

signal (11): Segmentation fault
in expression starting at REPL[17]:1
GMT_pslegend at /usr/lib/libgmt.so (unknown line)
GMT_Call_Module at /usr/lib/libgmt.so (unknown line)
GMT_Call_Module at /home/admin/.julia/packages/GMT/jguOm/src/libgmt.jl:147 [inlined]
gmt at /home/admin/.julia/packages/GMT/jguOm/src/gmt_main.jl:292
finish_PS_module at /home/admin/.julia/packages/GMT/jguOm/src/common_options.jl:3533
finish_PS_module at /home/admin/.julia/packages/GMT/jguOm/src/common_options.jl:3473 [inlined]
#legend#397 at /home/admin/.julia/packages/GMT/jguOm/src/pslegend.jl:65
legend##kw at /home/admin/.julia/packages/GMT/jguOm/src/pslegend.jl:51 [inlined]
#legend#403 at /home/admin/.julia/packages/GMT/jguOm/src/pslegend.jl:219 [inlined]
legend##kw at /home/admin/.julia/packages/GMT/jguOm/src/pslegend.jl:219
unknown function (ip: 0x7efc0d528b56)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2247 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2429
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1788 [inlined]
do_call at /buildworker/worker/package_linux64/build/src/interpreter.c:126
eval_value at /buildworker/worker/package_linux64/build/src/interpreter.c:215
eval_stmt_value at /buildworker/worker/package_linux64/build/src/interpreter.c:166 [inlined]
eval_body at /buildworker/worker/package_linux64/build/src/interpreter.c:587
jl_interpret_toplevel_thunk at /buildworker/worker/package_linux64/build/src/interpreter.c:731
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:885
jl_toplevel_eval_flex at /buildworker/worker/package_linux64/build/src/toplevel.c:830
jl_toplevel_eval_in at /buildworker/worker/package_linux64/build/src/toplevel.c:944
eval at ./boot.jl:373 [inlined]
eval_user_input at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/REPL/src/REPL.jl:150
repl_backend_loop at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/REPL/src/REPL.jl:244
start_repl_backend at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/REPL/src/REPL.jl:229
#run_repl#47 at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/REPL/src/REPL.jl:362
run_repl at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.7/REPL/src/REPL.jl:349
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2247 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2429
#930 at ./client.jl:394
jfptr_YY.930_40362.clone_1 at /usr/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2247 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2429
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1788 [inlined]
jl_f__call_latest at /buildworker/worker/package_linux64/build/src/builtins.c:757
#invokelatest#2 at ./essentials.jl:716 [inlined]
invokelatest at ./essentials.jl:714 [inlined]
run_main_repl at ./client.jl:379
exec_options at ./client.jl:309
_start at ./client.jl:495
jfptr__start_40531.clone_1 at /usr/lib/julia/sys.so (unknown line)
_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2247 [inlined]
jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2429
jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1788 [inlined]
true_main at /buildworker/worker/package_linux64/build/src/jlapi.c:559
jl_repl_entrypoint at /buildworker/worker/package_linux64/build/src/jlapi.c:701
main at julia (unknown line)
__libc_start_main at /usr/bin/../lib/libc.so.6 (unknown line)
unknown function (ip: 0x400808)
Allocations: 12158722 (Pool: 12153201; Big: 5521); GC: 14

^C
zsh: segmentation fault (core dumped)  julia --banner=no --project=$PWD
```

**Comment by adigitoleo** [comment]
But adding any kind of `width` parameter prevents the crash and the legend is plotted correctly.

**Comment by adigitoleo** [comment]
Seems to be two issues: the missing `width` segfault and inability to read from a legend file:

```
julia> gmtbegin("test_legend.png")

shell> cat legend.txt
S 0.25c - 0.33 black 2p 0.75c 1s delay time
S 0.25c - 0.33 black 2p 0.75c 1s delay time

julia> legend("legend.txt", position = (; inside = :BL, anchor = :BL, width = 5), box = (; pen = 0.5), region = (0, 10, 0, 8))
legend [WARNING]: File <stdin> is empty!

julia> gmtend(show = true) # No legend
```

**Comment by adigitoleo** [comment]
The file reading issue persists in old GMT mode:

```
julia> using GMT

shell> cat legend.txt
S 0.25c - 0.33 black 2p 0.75c 1s delay time
S 0.25c - 0.33 black 2p 0.75c 1s delay time

julia> legend("legend.txt", position = (; inside = :BL, anchor = :BL, width = 5), box = (; pen = 0.5), region = (0, 10, 0, 8))
pslegend [WARNING]: File <stdin> is empty!

julia> showfig()
```

**Comment by joa-quim** [comment]
Yep, I can see that in ``pslegend.jl`` the ``cmd0`` arg that contains the file name is only used in ``monolithic`` mode.
This one should be easy to fix.

**Comment by joa-quim** [comment]
Fixed in #829

**Comment by adigitoleo** [comment]
Thanks, it works.
